compiler/noirc_evaluator/src/ssa/ir/instruction/binary.rs

```diff
@@ -141,6 +141,23 @@ impl Binary {
                     let zero = dfg.make_constant(FieldElement::zero(), operand_type);
                     return SimplifyResult::SimplifiedTo(zero);
                 }
+                if operand_type.is_unsigned() {
+                    // lhs % 2**bit_size is equivalent to truncating `lhs` to `bit_size` bits.
+                    // We then convert to a truncation for consistency, allowing more optimizations.
```

Comment:
This raises the question of whether we need a Truncate instruction at all or if we should switch to only Mod instead. They seem to do the same task now with the only differences being Mod is more flexible and Truncate has `max_bit_size`.

Reply:
Yeah, that thought has come up to me as well. One benefit of keeping it are that having a power of 2 as a modulus is an important special case so it's easier to do some optimisations. Open to discussion on this.

Reply:
#4347